### ISSUE TITLE
feat(native): add GPT-SoVITS TTS via the Genie-TTS ONNX path (CPU-only card)

### ISSUE THREAD

**Issue by jiangzhuo** [kizuna-ai-lab/sokuji#322]
Add **GPT-SoVITS** (v2ProPlus) as a Local Native TTS card, running torch-free through the [Genie-TTS](https://github.com/High-Logic/Genie-TTS) ONNX pipeline, **deliberately shipped as a CPU-only card**.

Policy context: model candidates are judged on hard conditions only (torch-free runtime, commercial+AGPL-compatible license, real-time on ≥1 tier) — overlap with existing cards is not a factor.

## Eligibility (verified 2026-07-16)

| Condition | Status |
|---|---|
| License | ✅ **MIT** end-to-end: [GPT-SoVITS code](https://github.com/RVC-Boss/GPT-SoVITS/blob/main/LICENSE), [pretrained weights](https://huggingface.co/lj1995/GPT-SoVITS), [chinese-hubert-base](https://huggingface.co/TencentGameMate/chinese-hubert-base), Genie-TTS engine. RoBERTa (optional, zh prosody) is Apache-2.0. |
| Torch-free | ✅ `genie_tts` 2.0.2 wheel has **no torch** in Requires-Dist: onnxruntime, onnx, tokenizers, numpy, soundfile, soxr + pure-Python G2P (pyopenjtalk-plus, pypinyin, g2pM, jieba_fast, nltk). All stages have ONNX: cn-hubert (188.7 MB fp16), speaker encoder (184.8 MB), optional RoBERTa (599.5 MB), prompt_encoder / t2s_encoder / t2s_fsdec / t2s_sdec / vits (~337 MB per voice-set). |
| Real-time | ✅ on CPU: ~13 ms per AR step (25 Hz semantic tokens → 1 step = 40 ms audio) ⇒ RTF ≈ 0.3 on a decent laptop CPU (i7-13620H: 1.13 s for a ~3–3.5 s JA sentence), ~0.8–1.0 on weak CPUs. |

## Why CPU-only, and how to mark it

The Genie author states ORT-GPU gives this architecture no benefit ([Genie-TTS#38](https://github.com/High-Logic/Genie-TTS/issues/38)): *"onnxruntime-gpu 推理此类模型性能很差(相比 CPU 版无显著优势)，TRT 直接推理不太现实"*. Benchmarks agree: ORT on an RTX 2080Ti lands at the same ~13 ms/step as a laptop CPU — the AR loop is launch-bound, not compute-bound.

**Marking a card CPU-only requires no flag and no planner changes** — device support is expressed by which `Deployment` tiers exist on the card (`sidecar/sokuji_sidecar/catalog.py`). Give it a single row:

```python
Deployment("gpt_sovits_onnx", "cpu", "fp32", <artifact>, 1.0)
```

`planner.resolve_deployments` (planner.py:155) only plans over rows that exist, so this card can never land on GPU. Established precedent: every `_sherpa_tts_row` card is CPU-only exactly this way (catalog.py — "CPU-only by reality" comment), and Supertonic 3 has no gpu-metal row for the same reason. The renderer's tier tooltip derives from the same rows, so the UI will correctly show cpu only.

## Integration work

1. **New backend `gpt_sovits_onnx`** in `tts_backends.py` — **vendor** the Genie runtime rather than pip-install it: it pins `onnxruntime==1.22.1` (CPU wheel), which is mutually exclusive with our `onnxruntime-gpu`, and hardcodes `CPUExecutionProvider` (harmless here — the card is CPU-only by design). MIT allows vendoring.
2. **Offline conversion + hosting**: Genie's converter is not `torch.onnx.export` — it injects `.pth`/`.ckpt` tensors into pre-built ONNX graph templates (`torch.load` only, offline). Run once against the base pretrained checkpoints (`s1v3.ckpt` 155.3 MB + `s2Gv2ProPlus.pth` 200.1 MB + `sv/eres2netv2` 107.5 MB from [lj1995/GPT-SoVITS](https://huggingface.co/lj1995/GPT-SoVITS)) and host the result on our HF org (same pattern as qwen3-tts).
   ⚠️ **Spike this first** — the converter routing on *base* (non-fine-tuned) checkpoints is the one structurally-unverified step in the research.
3. **Catalog row**: `languages=("zh","en","ja")`, `clones=True`, `transcript_required=True` (reference clip + its transcript, the Qwen3-TTS UX — `ReferenceAudio(prompt_wav, prompt_text, language)`), `streaming=False`, ~750 MB (en/ja) or ~1.35 GB (with zh RoBERTa prosody). Decide whether RoBERTa ships as part of the card or as an optional download.
4. **Required guards** (known upstream bugs that hit our exact usage):
   - **Short-text bug**: 1–2 char inputs make Genie return/read the *reference clip* instead of the target text — live translation emits short segments constantly. Guard in the backend (min-length check / skip / merge with next segment).
   - English quality issues are reported upstream; validate before flagging `recommended`.
5. **Voice UX**: fits the existing two-axis voice capability (`custom="clip"` + `transcriptRequired`) — no renderer changes expected.
6. Tests mirroring the existing TTS backend tests.

## Known limitations (accepted at filing time)

- CPU-only (by design, see above); TTFA ~1.1–1.2 s; sentence-level chunking, no intra-utterance streaming; zh/en/ja only (Korean deps exist upstream but are not in the released wheel); zero-shot is GPT-SoVITS's weaker mode (its fame is few-shot fine-tuning, which is out of scope).

## References

- Genie-TTS: https://github.com/High-Logic/Genie-TTS · ONNX assets: https://huggingface.co/High-Logic/Genie
- GPT-SoVITS: https://github.com/RVC-Boss/GPT-SoVITS · weights: https://huggingface.co/lj1995/GPT-SoVITS
- ORT benchmarks: https://github.com/GPT-SoVITS-Devel/GPT-SoVITS-cpp (Python-ORT RTF 0.33 / C++-ORT 0.24 on 2080Ti)
- Catalog mechanism: `sidecar/sokuji_sidecar/catalog.py` (`Deployment.tier`), `sidecar/sokuji_sidecar/planner.py` (`resolve_deployments`, `TIER_RANK`)
